src/main/java/xbot/common/controls/actuators/XCANMotorController.java

```diff
@@ -223,33 +235,212 @@ public void periodic() {
 
     public abstract void setPowerRange(double minPower, double maxPower);
 
-    public Angle getPosition() {
+    public void setDistancePerAngleScaleFactor(Measure<? extends PerUnit<DistanceUnit, AngleUnit>> distancePerAngle) {
+        this.distancePerAngleScaleFactor = distancePerAngle;
+    }
+
+    public void setAngleScaleFactor(Measure<? extends PerUnit<AngleUnit, AngleUnit>> angleScaleFactor) {
+        this.angleScaleFactor = angleScaleFactor;
+    }
+
+    /**
+     * Get the position reported by the motor controller.
+     * @apiNote Angle scaling factors configured on the motor controller are ignored.
+     * @return The position reported by the motor controller.
+     */
+    public Angle getRawPosition() {
         return inputs.angle;
     }
 
+    /**
+     * Get the position reported by the motor controller.
+     * @apiNote Distance per angle scaling factors configured on the motor controller are applied.
+     * @return The position reported by the motor controller.
+     */
+    public Distance getPositionAsDistance() {
+        if (distancePerAngleScaleFactor == null) {
+            log.warn("Distance per angle not set for motor controller {}", akitName);
+            return Meters.zero();
+        }
+        return getRawPosition().timesConversionFactor(distancePerAngleScaleFactor);
```

Comment:
⭐ in addition to scaling, we almost always have a 'valueAtZero' offset to subtraction from the raw position, is that something we should maybe bake into this as well? Or are you thinking that we'd start tracking that offset at zero in scaled Distance rather than raw positions? Or should we be using setPosition more to zero encoders that way?

Reply:
SetPosition to move the zero seems like the cleanest way to avoid extra math.